cleanlab_studio/studio/inference.py

```diff
@@ -45,13 +48,27 @@ def predict(
         Returns:
             predictions from batch as a numpy array, optionally also pandas dataframe of class probabilties
         """
-        csv_batch = self._convert_batch_to_csv(batch)
-        predictions, class_probabilities = self._predict_from_csv(csv_batch, timeout)
+
+        batchsize: int = BATCH_MAX_TABULAR if isinstance(batch, pd.DataFrame) else BATCH_MAX_TEXT
+        batched_preds: List[Predictions] = []
+        batched_probs: List[ClassProbablities] = []
+        num_batches = len(batch) // batchsize + (1 if len(batch) % batchsize != 0 else 0)
+        for i in range(num_batches):
+            start = i * batchsize
+            csv_batch = self._convert_batch_to_csv(
+                batch[start : min(start + batchsize, len(batch))]
+            )
+            predictions, class_probabilities = self._predict_from_csv(csv_batch, timeout)
+            batched_preds.append(predictions)
+            batched_probs.append(class_probabilities)
+        preds_type = int if isinstance(batched_preds[0][0], int) else str
```

Comment:
nit: this might be clearer if written as `preds_type = type(batched_preds[0][0])` but not too important

Reply:
I think that using `type` might be hinky because type() doesn't respect inheritance but isinstance() does and we have like pd.Int64 and idk different numpy strings that give me the heebie jeebies

Reply:
okay i see, that makes sense. let's keep it as-is right now